### DISCUSSION TITLE
Checking dirty status on nested fields

### DISCUSSION THREAD

**Discussion by noahsb** [mantinedev/mantine#3192] [Q&A]
Curious if there is a simple way to check a nested fields dirty status without checking every single index?

Example:

```
intitialValues: {
  books: [
    { name: 'Harry Potter', author: 'J. K. Rowling' },
    { 'name': 'Little Women', author: 'Louisa May Alcott' }
  ]
}

const isNameDirty = form.isDirty('books.0.name') || form.isDirty('books.1.name')
```

I'd prefer to not do this manually or create a loop to check this every time if there's a utility to do this. 

Thanks for all your amazing work maintaining this library!

**Comment by rtivital** [comment]
Why `form.isDirty('books')` does not work for you?

**Reply by noahsb** [reply]
I have a unique use case where I only care about the dirty status of `name` and not `author`. I want to perform a specific action if _any_ `name` is dirty. 

So:

```
if (isNameDirty) {
  // perform side effect
}
```

**Reply by rtivital** [reply]
Well, not much you can do then atm. If you are interested, you can create a PR that will add index reference feature in this fucntion, should be the same as in here – https://mantine.dev/form/validation/#validate-fields-on-change
